start_proxy.sh

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+err() {
+	echo -e "\e[31m[!]\e[0m ${1}" 1>&2
+}
+
+inf() {
+	echo -e "\e[34m[*]\e[0m ${1}"
+}
+
+leave() {
+	[ -f /etc/hosts.bak ] && { inf "Fixing /etc/hosts..." && sudo mv /etc/hosts.bak /etc/hosts; }
+
+	[ -z ${lldb_pid+x} ] || { inf "Killing attached lldb..." && { sudo kill "$lldb_pid" 2>/dev/null || :; }; }
+	[ -z ${mitm_pid+x} ] || { inf "Killing mitmweb..." && { kill "$mitm_pid" 2>/dev/null || :; }; }
+	[ -z ${proxy_pid+x} ] || { inf "Killing proxy..." && { kill "$proxy_pid" 2>/dev/null || :; }; }
+
+	cd "$old_dir" || :
+
+	exit 0
+}
+
+[[ "$(uname)" != "Darwin" ]] && { err "This can only be run on macOS" && exit 1; }
+
+old_dir="$(pwd)"
+root_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+full_dir="$root_dir"
+[[ "$old_dir" == "$root_dir" ]] && root_dir="."
+
+trap 'leave' INT ERR
+
+proxy_dir="${root_dir}/development/proxy"
+
+inf "Setting up \e[1mhosts.proxy\e[0;34m..."
+
+hosts_proxy="${proxy_dir}/hosts.proxy"
+cat /etc/hosts > "$hosts_proxy"
+python3 "${proxy_dir}/hosts.py" >> "$hosts_proxy"
+
+echo -e "\e[32m[?]\e[0;1m ${hosts_proxy}\e[0m must be copied over to /etc/hosts. Would you like us to do that for you? [y/n]"
+read -rn1 answer
+if [[ "${answer,,}" == "y" ]] 
+then
+	inf "Backing up /etc/hosts to /etc/hosts.bak and copying ${hosts_proxy} to /etc/hosts"
+	sudo cp /etc/hosts /etc/hosts.bak
+	sudo cp "$hosts_proxy" /etc/hosts
+fi
+
+lldb_commands="${proxy_dir}/lldb_commands.txt"
+cat << EOF > "$lldb_commands"
+breakpoint set -n "SecTrustEvaluateWithError" -C "thread return 1" -C "c"
+c
+EOF
+
+inf "Attaching to lldb..."
+env TERM=xterm-256color sudo lldb -p $(pgrep apsd) -s "$lldb_commands" >/dev/null &
+lldb_pid=$!
+
+cat << EOF > "${proxy_dir}/imessage_proxy.pac"
+// https://en.wikipedia.org/wiki/Proxy_auto-config for reference
+function FindProxyForURL(url, host) {
+  // to redirect apns tcp traffic
+  if (shExpMatch(host, '*-courier.push.apple.com')) {
```

Comment:
why are you putting courier.push.apple.com through the HTTP proxy? Only APNs traffic should be connecting to that domain, so it would not do anything?

Reply:
yeah that's probably not really necessary; i put this in there when I was trying to see if it would proxy anything other than http(s) and ftp, and didn't remove it when i realized that's not the case. i could remove it, but since it's another domain that we know has something to do with apple/imessage, i don't see harm in leaving it just in case it happens to reveal something, i guess?